sdk/messaging/azeventgrid/example_test.go

```diff
@@ -4,20 +4,139 @@
 package azeventgrid_test
 
 import (
+	"context"
+	"fmt"
+	"log"
 	"os"
 
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/messaging"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
 	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventgrid"
 )
 
 func ExampleNewClientWithSharedKeyCredential() {
 	endpoint := os.Getenv("EVENTGRID_ENDPOINT")
 	sharedKey := os.Getenv("EVENTGRID_KEY")
 
+	if endpoint == "" || sharedKey == "" {
+		return
+	}
+
 	client, err := azeventgrid.NewClientWithSharedKeyCredential(endpoint, sharedKey, nil)
 
 	if err != nil {
-		panic(err)
+		//  TODO: Update the following line with your application specific error handling logic
+		log.Fatalf("ERROR: %s", err)
 	}
 
 	_ = client // ignore
+
+	// Output:
+}
+
+func ExampleClient_PublishCloudEvents() {
+	client := getEventGridClient()
```

Comment:
I wonder how this will render on `pkg.go.dev`.  It might end up being that the examples aren't copy/pasteable.

Reply:
I use pkgsite - they render nicely! It actually brings in the function and the called function in the file.

Reply:
I should mention - I could have sworn it didn't used to work well either, so I was pleasantly surprised.